tests/test_validate_dummy_data.py

```diff
@@ -78,3 +78,28 @@ def test_validate_dummy_data_missing_data_file(file_format):
             fixtures_path / f"missing.{file_format}",
             Path(f"output.{file_format}"),
         )
+
+
+@pytest.mark.parametrize("default_value", ["missing", None])
+def test_validate_dummy_data_with_categories(tmpdir, default_value):
+    class CohortWithCategories:
+        population = table("practice_registations").exists()
+        _code = table("clinical_events").filter(code__in=cl)
+        event_date = _code.latest().get("date")
+        _categories = {
+            1: event_date == "2021-01-01",
+        }
+        category = categorise(_categories, default=default_value)
```

Comment:
I did old-style DSL here so as not to mix up old and new, but maybe it should be new-style instead :shrug: